### ISSUE TITLE
Shortcomings in the example autoencoder

### ISSUE THREAD

**Issue by ramayer** [ramayer/hierarchical-binary-quantization#29]
From a claude code review comparing the two example autoencoders: 

| Feature| MinimalSpatialViT | LatentJiT | Impact |
| ---|---|---|---|
| Linear layer bias | The CS336 assignment implemented a linear layer with no bias | uses `nn.Linear(..., bias=True)` | Huge language models prefer to omit bias "for memory ordering reasons", appropriate for LLMs built on RMSNorm.    For this model, the lack of a bias means the network has to learn how to model constant per-channel offsets, through RMSNorm-normalized (near-zero-mean) activations which can be slow to fit.   Biases would otherwise absorb most of the "obvious" signal (flow-matching targets rarely have zero mean per channel). |
| QK-Norm | raw q/k fed straight into RoPE | RMSNorm applied to q and k per-head before RoPE (qk_norm=True) | QK Norm stablizes attention in many models (ViT-22B, Gemma, and JiT).  |
| Absolute position embedding | Non - RoPE Only | Fixed 2D sin-cos pos_embed added in addition to RoPE | Combining both gives an easeir optimization path [but in this case I think we do NOT want the feature; because while it'll be faster to learn "a head in the top center of an image", it'll be slower to learn "people may not always be centered | 
| Final Layer Zero Init | Only adaLN (per-block) was zero-inited | adaLN and final_layer.linear weight and bias were zero inited ; final_linear had ordinatry trunc_normal init| Jit and DiT start by predicting zero, so early loss is close to a known good solution (prediction v=0 in flow matching).   First hundred steps of the minimal spatial vit needs to re-learn that fact |
| Rope 2D construction | Split-half per-axis RoPE applied at forward-time via row/col lookup | precomputed grid broadcast freqs in VisionRotaryEmbeddingFast | Likely equivalent |
| Norm Type | RMSNorm | RMSNorm | Same |
| FFN | SwiGLU dff=8/3*d_modal, no bias | SwiGLU, dff = 2/3*hidden * mlp_ratio, bias = True | Similar |
| AdaLN | 6*d_model per block | 6*hidden_size per block | Same |
